src/components/MapComponent.tsx

```diff
@@ -133,17 +137,21 @@ export const MapComponent: React.FC<Props> = ({ options, width, height, data, on
                       key={index}
                       positions={[point, linePoints[index + 1]]}
                       weight={5}
-                      opacity={1 / (linePoints.length / (index + 2))}
-                      color={'green'}
+                      opacity={0.6}
+                      color={colorValues[index]}
                     />
                   );
                 })}
             </>
           );
         })}
-      <div className="leaflet-bottom leaflet-right">
-        <button className="leaflet-control" onClick={() => setHistoryVisibility(!historyVisibility)}>
-          click
+      <div className="leaflet-bottom leaflet-left">
```

Comment:
I added tail visibility to the map options so that the visibility information is saved in Grafana's state and remains unchanged after refreshing the page.